CRM/Report/Form/Contribute/Detail.php

```diff
@@ -588,7 +588,7 @@ public function postProcess() {
     $select = str_ireplace('contribution_civireport.total_amount', 'contribution_soft_civireport.amount', $this->_select);
     $select = str_ireplace("'Contribution' as", "'Soft Credit' as", $select);
     // We really don't want to join soft credit in if not required.
-    if (!empty($this->_groupBy) && !$this->noDisplayContributionOrSoftColumn) {
+    if (!empty($this->_groupBy) && $this->isTableSelected('contribution_soft_civireport')) {
```

Comment:
Without this patch when I run the report I get 

Notice: Undefined property: CRM_Report_Form_Contribute_Detail::$noDisplayContributionOrSoftColumn in CRM_Report_Form_Contribute_Detail->postProcess() (line 591 of /Users/emcnaughton/buildkit/build/dmaster/sites/all/modules/civicrm/CRM/Report/Form/Contribute/Detail.php).

Reply:
hmmm annoying